### PR TITLE
UCP/MM/TEST: Allow mmap with address=NULL and length=0

### LINKED ISSUE #2588: invalid error check in ucp_mem_map
when ucp_mem_map is called with (addr=NULL, length=0, flags=0)
it should return dummy_mem_handle and not assume user is trying to allocate memory
see also https://github.com/open-mpi/ompi/issues/5083#issuecomment-383094506

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #2588

### CODE REVIEW

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -966,7 +966,7 @@ typedef struct ucp_mem_map_params {
       * Length (in bytes) to allocate or map (register).
       * This field is mandatory for filling (along with its corresponding bit
       * in the field_mask - @ref UCP_MEM_MAP_PARAM_FIELD_LENGTH).
-      * The @ref ucp_mem_map routine will return with an error if the length isn't
```
Comment (brminich) [inline src/ucp/api/ucp.h]: typo?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7000/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@gmegan pls take a look

**Comment by yosefe** [comment]
adding @shamisp

**Pull request merged**